kong/tools/request_aware_table.lua

```diff
@@ -0,0 +1,72 @@
+--- NOTE: tool is designed to assist with **detecting** request contamination
+-- issues on CI, during test runs. It does not offer security safeguards.
+
+local table_new = require "table.new"
+local base = require "resty.core.base"
+
+local debug_mode  = kong.configuration.log_level == "debug"
```

Comment:
I would like to improve this so that it cannot be enabled by anyone who's running with debug log level. I think we may be able to do better and make sure the table is only patched in integration tests, for example by using some logic based on flags that are passed by the helpers when calling start_kong, similarly to what we do [here](https://github.com/Kong/kong/blob/613fb31961565947dd8873c8175040447f0ffb44/spec/helpers.lua#L3588).

Reply:
It is probably not very necessary, we don't let user run debug mode in production anyway and the risk should be minimum.